aiida_vasp/parsers/base.py

```diff
@@ -90,6 +91,7 @@ def _check_folder(self):
         except NotExistent:
             return self.exit_codes.ERROR_NO_RETRIEVED_FOLDER
 
+    @contextmanager
```

Comment:
If I understand correctly, context manager is used to replace try and finally, but not try and except. So I feel the code is written not very natural.
In addition, the outer 'try' may be avoided, which probably makes the code more readable, though I don't see what you aim here exactly.

Reply:
I guess that here the intention is to allow `None` to be yield in case the file does not exist. 

I agree with @atztogo  that the `try..except`` is a bit unnecessary, we can just check if the key exists and act accordingly, otherwise we might end up suppressing exception which should have been raised.

Reply:
I have not really started in the `parsers` folder yet. Getting the `content_parsers` working first. But I agree.